@radcortez, I think this is creating some config confusion:

```
2025-01-28 16:39:00,816 INFO  [io.quarkus] (Quarkus Main Thread) quarkus-vertx-http-deployment 999-SNAPSHOT on JVM (powered by Quarkus 999-SNAPSHOT) started in 2.918s. Listening on: http://localhost:8080
2025-01-28 16:39:00,819 INFO  [io.quarkus] (Quarkus Main Thread) Profile dev activated. Live Coding activated.
2025-01-28 16:39:00,819 INFO  [io.quarkus] (Quarkus Main Thread) Installed features: [cdi, security, smallrye-context-propagation, vertx]
2025-01-28 16:39:04,033 INFO  [io.quarkus] (Quarkus Main Thread) quarkus-vertx-http-deployment stopped in 0.007s
Error:  Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 9.100 s <<< FAILURE! -- in io.quarkus.devui.BuildMetricsTest
Error:  io.quarkus.devui.BuildMetricsTest.testGetBuildStepsMetrics -- Time elapsed: 8.933 s <<< ERROR!
java.lang.IllegalStateException: Request failed: Connection refused: localhost/127.0.0.1:8081
	at io.quarkus.devui.tests.DevUIJsonRPCTest$WebSocketResponse.message(DevUIJsonRPCTest.java:254)
	at io.quarkus.devui.tests.DevUIJsonRPCTest.objectResultFromJsonRPC(DevUIJsonRPCTest.java:155)
	at io.quarkus.devui.tests.DevUIJsonRPCTest.objectResultFromJsonRPC(DevUIJsonRPCTest.java:167)
	at io.quarkus.devui.tests.DevUIJsonRPCTest.objectResultFromJsonRPC(DevUIJsonRPCTest.java:148)
	at io.quarkus.devui.tests.DevUIJsonRPCTest.getJsonRPCResponse(DevUIJsonRPCTest.java:124)
	at io.quarkus.devui.tests.DevUIJsonRPCTest.getJsonRPCResponse(DevUIJsonRPCTest.java:119)
	at io.quarkus.devui.tests.DevUIJsonRPCTest.executeJsonRPCMethod(DevUIJsonRPCTest.java:86)
	at io.quarkus.devui.tests.DevUIJsonRPCTest.executeJsonRPCMethod(DevUIJsonRPCTest.java:75)
	at io.quarkus.devui.tests.DevUIJsonRPCTest.executeJsonRPCMethod(DevUIJsonRPCTest.java:71)
	at io.quarkus.devui.BuildMetricsTest.testGetBuildStepsMetrics(BuildMetricsTest.java:23)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1511)
Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: localhost/127.0.0.1:8081
Caused by: java.net.ConnectException: Connection refused
	at java.base/sun.nio.ch.Net.pollConnect(Native Method)
	at java.base/sun.nio.ch.Net.pollConnectNow(Net.java:672)
	at java.base/sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:946)
	at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:336)
	at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:339)
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:776)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724)
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650)
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562)
	at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997)
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74)
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
	at java.base/java.lang.Thread.run(Thread.java:840)
```

The server is started on port 8080 but the tests try to connect on 8081.

--- a/core/runtime/src/main/java/io/quarkus/runtime/configuration/QuarkusConfigFactory.java
+++ b/core/runtime/src/main/java/io/quarkus/runtime/configuration/QuarkusConfigFactory.java
@@ -1,7 +1,5 @@
 package io.quarkus.runtime.configuration;
 
-import org.eclipse.microprofile.config.spi.ConfigProviderResolver;
-
 import io.quarkus.runtime.LaunchMode;
 import io.smallrye.config.SmallRyeConfig;
 import io.smallrye.config.SmallRyeConfigFactory;
@@ -25,7 +23,8 @@ public final class QuarkusConfigFactory extends SmallRyeConfigFactory {
     }
 
     public static void setConfig(SmallRyeConfig config) {
-        ConfigProviderResolver configProviderResolver = ConfigProviderResolver.instance();
+        SmallRyeConfigProviderResolver configProviderResolver = (SmallRyeConfigProviderResolver) SmallRyeConfigProviderResolver
+                .instance();
         // Uninstall previous config
         if (QuarkusConfigFactory.config != null) {
             configProviderResolver.releaseConfig(QuarkusConfigFactory.config);
@@ -34,9 +33,8 @@ public final class QuarkusConfigFactory extends SmallRyeConfigFactory {
         // Install new config
         if (config != null) {
             QuarkusConfigFactory.config = config;
-            // Register the new config for the TCCL,
-            // just in case the TCCL was using a different config
-            // than the one we uninstalled above.
+            // Someone may have called ConfigProvider.getConfig which automatically registers a Config in the TCCL
+            configProviderResolver.releaseConfig(Thread.currentThread().getContextClassLoader());
             configProviderResolver.registerConfig(config, Thread.currentThread().getContextClassLoader());
         }
     }

